### PR TITLE
Add isDawn flag to `InjectedProviderFlags`

### PR DESCRIPTION
**Pull request by thomas-waite**
## Description
Dawn Wallet is an iOS based Ethereum wallet with a native Safari extension: https://twitter.com/dawn_wallet. 

We are integrating into ConnectKit - ConnectKit uses the `Ethereum` type from wagmi, which itself uses the `InjectedProviderFlags` type to recognise injected wallets. 

We need the `isDawn` flag to be available on the `Ethereum` type in ConnectKit (see this PR: https://github.com/family/connectkit/pull/162). This PR in wagmi adds it.

Your ENS/address: tomwaite.eth

### CODE REVIEW

**Review by jxom — CHANGES_REQUESTED** [review]
Thanks for the PR!

Can you please add a patch `changeset` via `pnpm changeset`?

Will also need to add Dawn into [`getName`](https://github.com/wagmi-dev/references/blob/04106f41506fd7b856d638a13f1a710098b3b426/packages/connectors/src/utils/getInjectedName.ts#L6) and the [tests](https://github.com/wagmi-dev/references/blob/04106f41506fd7b856d638a13f1a710098b3b426/packages/connectors/src/utils/getInjectedName.test.ts#L6).

**Comment by thomas-waite** [comment]
> Thanks for the PR!
> 
> Can you please add a patch `changeset` via `pnpm changeset`?
> 
> Will also need to add Dawn into [`getName`](https://github.com/wagmi-dev/references/blob/04106f41506fd7b856d638a13f1a710098b3b426/packages/connectors/src/utils/getInjectedName.ts#L6) and the [tests](https://github.com/wagmi-dev/references/blob/04106f41506fd7b856d638a13f1a710098b3b426/packages/connectors/src/utils/getInjectedName.test.ts#L6).

Done! Let me know if anything else needs doing

**Comment by thomas-waite** [comment]
@jxom Lmk when this looks good to go, need to get this merged and a new package published for our connector PR in ConnectKit: https://github.com/family/connectkit/pull/162 

Thanks for reviewing!

File: .changeset/ninety-needles-greet.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@wagmi/connectors': patch
+---
+
+Add isDawn wallet flag
```
Comment (tmm) [inline .changeset/ninety-needles-greet.md:5]:
```suggestion
Added Dawn wallet flag
```

**Review by tmm — APPROVED** [review]

**Pull request merged**